packages/instantsearch.js/src/components/RangeInput/RangeInput.tsx

```diff
@@ -45,12 +45,14 @@ class RangeInput extends Component<RangeInputProps, Partial<Range>> {
     });
   }
 
-  private onInput = (key: string) => (event: Event) => {
+  private onInput = (key: keyof typeof this.state) => (event: Event) => {
     const { value } = event.currentTarget as HTMLInputElement;
 
-    this.setState({
-      [key]: Number(value),
-    });
+    if (value) {
+      this.setState({
+        [key]: Number(value),
+      });
+    }
```

Comment:
hm, that seems hard to detect. Does it actually have an impact, and is that what people actually want? eg if you remove the limit on max, shouldn't it go to the default max instead of 0 (thus behaviour of this PR I believe)

Reply:
oh, but this fix keeps the last number if you erase the input :(

Reply:
correct solution would probably be to not keep the value in state, and only set a `defaultValue`? I'll try it out later

Reply:
I don't think so, I just remembered having to accomodate this behavior in a previous contribution. I played a bit and indeed the value returned by the input is `undefined` with both empty string and invalid string (ending with separator). I'm fine if we keep it simple (leaving the input empty).

What bothers me more is that the input number type has a locale dependency on what should be a value separator, which we can't control reliably. People should be able to use dot `.` or comma `,`. Maybe we can switch to a text input with pattern validation in the next major or something, even if it's less semantically correct.

Reply:
Yes, I noticed that too, but couldn't find the right middle ground (as well as even though my computer is set up with comma instead of period, the value still only writes a period) 🤔

Reply:
I took a look at [FX-2274](https://algolia.atlassian.net/browse/FX-2274) with @aymeric-giraudet this morning.
If we keep the value as a `string`, it's working correctly (comma or dot), of course we don't do any validation this way, it's done by the browser, which force you to enter a valid number before submitting. We could cast it to number when we `refine`.

```suggestion
    this.setState({
      [key]: value,
    });
```

Reply:
ah, that's an interesting point! can you swap from string to number without warning as well? I think it's safe to change, feel free to take over the PR